src/dbg/TraceRecord.cpp

```diff
@@ -305,14 +305,14 @@ void TraceRecordManager::TraceExecuteRecord(const Zydis & newInstruction)
           )
         {
             MemRead(newContext.registers.regcontext.csp - sizeof(duint), &newMemory[newMemoryArrayCount], sizeof(duint));
-            newMemoryAddress[newMemoryArrayCount] = newContext.registers.regcontext.csp - sizeof(duint);
+            newMemoryAddress[--newMemoryArrayCount] = newContext.registers.regcontext.csp - sizeof(duint);

```

Comment:
While this works, I believe it would be more convenient if zydis was decoding operants as [?sp-size] instead of [?sp].